### PR TITLE
Sent out a status text for unknown HTTP headers.

### PR DESCRIPTION
**Pull request by dawehner**
- The HTTP RFC explains that header('HTTP/1.1 418 ') should be fine, see http://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html#sec6.1
- Apache itself though is broken, see https://gist.github.com/anonymous/a863d7b493c4b09733ec

The fix is to sent out some status text, when we don't know a better text.

| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by chx** [comment]
See https://bz.apache.org/bugzilla/show_bug.cgi?id=44995 and https://bz.apache.org/bugzilla/show_bug.cgi?id=54946 on why Apache refuses empty `Reason-Phrase`.

File: src/Symfony/Component/HttpFoundation/Response.php [hunk 1]
```diff
@@ -454,7 +454,7 @@ public function setStatusCode($code, $text = null)
         }
 
         if (null === $text) {
-            $this->statusText = isset(self::$statusTexts[$code]) ? self::$statusTexts[$code] : '';
+            $this->statusText = isset(self::$statusTexts[$code]) ? self::$statusTexts[$code] : 'Unknown status text';
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Response.php]: `Unknow Status` only (with caps)? text looks redondant

Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Response.php]: or maybe all lowercase to differenciate from regular status texts that are "ucfirst"

Comment (dawehner) [inline src/Symfony/Component/HttpFoundation/Response.php]: So let's go with 'unknown status'?

**Comment by fabpot** [comment]
Thank you @dawehner.

**Comment by Tobion** [comment]
Doesn't it say it's fixed in Apache?

**Pull request closed without merging**